File: content/snippets/go/language_basics/file_io.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	const path = "file_io_demo.txt"

	err := os.WriteFile(path, []byte("apple\nbanana\ncarrot\n"), 0o644)
	if err != nil {
		panic(err)
	}

	content, err := os.ReadFile(path)
	if err != nil {
		panic(err)
	}

	lines := strings.Fields(string(content))
	fmt.Println("read lines:", lines)
}

File: content/snippets/python/language_basics/file_io.py
#!/usr/bin/env python3
from pathlib import Path


def main() -> None:
    path = Path("file_io_demo.txt")
    path.write_text("apple\nbanana\ncarrot\n", encoding="utf-8")

    lines = path.read_text(encoding="utf-8").splitlines()
    print("read lines:", lines)


if __name__ == "__main__":
    main()
